src/matlab.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022-2024 The MathWorks, Inc.
+
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import * as http from "@actions/http-client";
+import * as io from "@actions/io";
+import * as tc from "@actions/tool-cache";
+import * as fs from "fs";
+import { homedir } from "os";
+import * as path from "path";
+import properties from "./properties.json";
+
+export interface Release {
+    name: string;
+    version: string;
+    update: string;
+}
+
+export async function makeToolcacheDir(release: Release, platform: string): Promise<[string, boolean]> {
+    let toolpath: string = tc.find("MATLAB", release.version);
+    let alreadyExists = false;
+    if (toolpath) {
+        core.info(`Found MATLAB ${release.name} in cache at ${toolpath}.`);
+        alreadyExists = true;
+    } else {
+        if (platform === "win32") {
+            toolpath = await windowsHostedToolpath(release).catch(async () => {
+                return await defaultToolpath(release, platform);
+            });
+        } else {
+            toolpath = await defaultToolpath(release, platform);
+        }
+    }
+    return [toolpath, alreadyExists]
+}
+
+async function windowsHostedToolpath(release: Release): Promise<string> {
+    // bail early if not on a github hosted runner
+    if (process.env['RUNNER_ENVIRONMENT'] !== 'github-hosted' && process.env['AGENT_ISSELFHOSTED'] === '1') {
```

Comment:
I don't know because it's not documented. This is how github does the checks in their internal actions. Because it's not documented, I felt most comfortable doing what they do.